plugins/discovery-ec2/src/main/java/org/opensearch/discovery/ec2/AwsEc2ServiceImpl.java

```diff
@@ -84,7 +84,7 @@ protected Ec2Client buildClient(
         ProxyConfiguration proxyConfiguration,
         ClientOverrideConfiguration overrideConfiguration,
         String endpoint,
-        Region region,
+        String region,
```

Comment:
Looking for some additional context here - why are we moving away from the typed reference to a string?

Reply:
It needs to default to null, doesn't look like settings can be `Optional` or `null`, at least I couldn't find an example. Do you know of one?

Reply:
Thanks. Yep I tried looking into it after I commented. 
We do not have `Optional` settings currently and a blank default value will make it blow up since `Region` does not have any auto-detect constants.